### ISSUE TITLE
Merge into mainline?

### ISSUE THREAD

**Issue by ViralBShah** [Keno/julia#27]
@loladiro I would like to start using some of these things in the mainline julia. 
1. Get rid of lighttpd, if ngnix works.
2. Get rid of fdlibm, and use openlibm.
3. Fix the libgfortran/libquadmath issue for distributing binaries, but you may be already addressing here.

Seems like you were waiting for some libuv related merges upstream. How about we keep using a forked version of libuv until upstream does what you need, and we get all the Windows stuff merged into the mainline?

**Comment by Keno** [comment]
1. I've been using both lighttpd and nginx (I managed to work around the issues with the former, though it is still rather unmaintained). Not sure what's the best way to go.
2. Yeah, definitely, though I think we're still using fdlibm in some places even on windows.
3. It's pretty common to package those on windows when using MinGw, so that's what I'm doing. Not sure about Unix-like platforms, though it should work as well.

Regarding libuv, I agree with you to keep our version of it separate and proceed with the merge in the meantime (I frequently talk to the libuv maintainers though they're quite busy at the moment). 
Regarding general progress, I think we're almost ready to merge into mainline and I'm flying back to Cambridge on Thursday, so I can come over to MIT sometime week or so to discuss any remaining issues and a proper merge strategy with @JeffBezanson and @StefanKarpinski in person.

**Comment by ViralBShah** [comment]
1. I'd rather do away with lighttpd given that it is not maintained. In fact, the whole web repl can be moved into a package, once we have packaging.
2. After the Windows merge, I will make sure we use openlibm everywhere on all platforms, and put our energies into improving it.
3. Even on unix platforms, we need to bundle libquadmath and libgfortran at the very least - especially on mac. On linux for standalone downloads, we should do that as well, although most distros should be able to do it with dependencies on the gfortran runtime.

-viral

**Comment by StefanKarpinski** [comment]
We should definitely ditch lighttpd in favor of nginx. Everyone uses nginx. I'm in NY right now and will be in CT this weekend, but after that I could come up to Cambridge. Really looking forward to meeting you, Keno!

**Comment by Keno** [comment]
Yeah, whenever you guys are available is fine. I'll be there for a while ;)

**Comment by Keno** [comment]
Just a heads up that I arrived Cambridge and got settled in, so I'm available anytime now.

**Comment by JeffBezanson** [comment]
Proposed date of merge is Oct. 2.

**Comment by Keno** [comment]
Is this still the plan? I could come over to MIT tomorrow and be there from ~5pm till around ~9pm, if you guys want to meet up to do this. (/cc @vtjnash)

**Comment by vtjnash** [comment]
@JeffBezanson Can you get us into Stata or pick another location? Or do we need to pick a new day? My evenings are surprisingly free this week.

**Comment by JeffBezanson** [comment]
Ok I will reserve a room in stata 5-9.

**Comment by JeffBezanson** [comment]
I have G631 tonight 5-9.

**Comment by vtjnash** [comment]
Result of this was that we should write a document describing the design of the new I/O API. (New I/O API is essentially blocking waits with callbacks and coroutines, as provided by libuv.)

**Comment by Keno** [comment]
What's left to do before we can merge this?

**Comment by JeffBezanson** [comment]
I'd love to see some documentation, since I am a bit overwhelmed at all the new types and function names in here. I suspect after a page of explanation it will all make sense.
